docs/code-generation/generate-hcl.md

```diff
@@ -129,19 +132,32 @@ terraform {
 }
 ```
 
-## tm_dynamic block
+In the next section, we will dive deeper into the process of working with `tm_dynamic` Blocks and explore practical examples.
+
+
+# Working with `tm_dynamic` Blocks
+
+The `tm_dynamic` block is a specialized block type that is utilized within the content block of the generate_hcl block. 
+
+It shares similarities with Terraform dynamic blocks but it also incorporates partial evaluation of the expanded code. This allows for more flexible and dynamic code generation.
+
+## Partial evaluation of Expanded Code
+
+One of the key features of the `tm_dynamic` block is its ability to perform partial evaluation of the expanded code. 
+
+This means that Terramate variables and functions within the block are evaluated, while other code elements are directly copied to the final generated code without evaluation.
 
-The `tm_dynamic` is a special block type that can only be used inside the
-`content` block of the `generate_hcl` block.
-It's similar to [Terraform dynamic blocks](https://www.terraform.io/language/expressions/dynamic-blocks)
-but supports partial evaluation of the expanded code.
+## Usage of Content and Attributes in `tm_dynamic` blocks
 
-The generate block's attributes can be provided by a `content` block, an
-`attributes` attribute, or even both if they don't conflict.
-When using the `content` block, additional sub-blocks can be generated and
-nested `tm_dynamic` blocks can be defined.
+The `tm_dynamic` block can be defined using either a content block, an attributes attribute, or a combination of both. 
 
-Example using the `content` block:
+The content block allows for the generation of additional sub-blocks and nesting of `tm_dynamic` blocks.
+
+## Examples: Generating Dynamic Blocks with `tm_dynamic`
+
+To get the more clear understanding of the concepts that we have explained above, let’s take some examples and understand those examples by looking at the generated outcome.
+
+### Using the Content Block:
```

Comment:
Why are we adding a colon here @Adeniyikayodee ? We don't do this in any of the other documentation sections?